src/ingest-pipeline/misc/tools/split_and_create.py

```diff
@@ -338,31 +369,36 @@ def reorganize(source_uuid, **kwargs) -> None:
     frozen_df_fname = kwargs["frozen_df_fname"]
     verbose = kwargs.get("verbose", False)
     dag_config = {}
+    child_uuid_list = []
+    is_multiassay = False
 
     entity_factory = EntityFactory(auth_tok, instance=instance)
 
     print(f"Decomposing {source_uuid}")
     source_entity = entity_factory.get(source_uuid)
+    full_entity = SoftAssayClient(
+        list(source_entity.full_path.glob("*metadata.tsv")), instance, auth_tok
+    )
     if mode in ["all", "stop"]:
-        if hasattr(source_entity, "data_types"):
+        if hasattr(source_entity, "dataset_type"):
```

Comment:
I think this should be fine, uploads should have a dataset_type specified.

Reply:
They should, that's generated in the ingest-ui (for uploads)